libp2p/directSender.go

```diff
@@ -30,21 +30,24 @@ const maxMutexes = 10000
 const sequenceNumberSize = 8
 
 type directSender struct {
-	counter         uint64
-	ctx             context.Context
-	hostP2P         host.Host
-	messageHandler  func(msg *pubsub.Message, fromConnectedPeer core.PeerID) error
-	mutSeenMessages sync.Mutex
-	seenMessages    *timecache.TimeCache
-	mutexForPeer    *MutexHolder
-	signer          p2p.SignerVerifier
+	counter           uint64
+	ctx               context.Context
+	hostP2P           host.Host
+	mutMessageHandler sync.RWMutex
+	messageHandler    p2p.MessageProcessor
+	mutSeenMessages   sync.Mutex
+	seenMessages      *timecache.TimeCache
+	mutexForPeer      *MutexHolder
+	signer            p2p.SignerVerifier
+	marshaller        p2p.Marshaller
 }
 
 // NewDirectSender returns a new instance of direct sender object
 func NewDirectSender(
 	ctx context.Context,
 	h host.Host,
 	signer p2p.SignerVerifier,
+	marshaller p2p.Marshaller,
```

Comment:
optional: we can think on adding args struct here

Reply:
Agree, can be made in a following PR

Reply:
right, will be added in a further PR